feature(add): Adding events (#47)
botpress/documentation: pages/integration/advanced/events.mdx

### Example:
Here's an illustration of how to define a new order event for Shopify:

```javascript
import { IntegrationDefinitionProps } from '@botpress/sdk'
import z from 'zod'

import { INTEGRATION_NAME } from '../const'

export type NewOrderCreated = z.infer<typeof newOrderCreated.schema>

const newOrderCreated = {
  schema: z.object({
    type: z.literal(`${INTEGRATION_NAME}:newOrderCreated`).optional(),
    order_id: z.number(),
    shopName: z.string(),
    confirmation_number: z.string(),
    created_at: z.string(),
    currency: z.string().optional(),
    current_subtotal_price: z.string().optional(),
    current_total_discounts: z.string().optional(),
    current_total_price: z.string().optional(),
    current_total_tax: z.string().optional(),
    customer_locale: z.string().optional(),
    order_status_url: z.string().optional(),
    fullBody: z.object({}).passthrough(),
  }),
  ui: {},
}

export const events = {
  newOrderCreated,
} satisfies IntegrationDefinitionProps['events']
```

## Event Implementation

> This refers to the actual code segment that contains the implementation of the action in the integration. The parameters that can be passed to the implementation function can include but are not limited to ctx (context), input (user-provided inputs in the fields defined in the UI section in the definition), logger (useful for logging information to the bot logs), and client (information about the integration).

Reviewer: ```suggestion
This refers to the actual code segment that contains the implementation of the action in the integration. The parameters that are passed to the implementation function include but are not limited to `ctx` (context), `input` (user-provided inputs in the fields defined in the UI section in the definition), `logger` (useful for logging information to the bot logs), and `client` (Botpress API Client).
```
